cmd/lib/api.go

```diff
@@ -47,13 +47,15 @@ func GetFullNodeApi(ctx context.Context, ai string, log *logging.ZapEventLogger)
 		return nil, nil, fmt.Errorf("creating full node service API: %w", err)
 	}
 
-	v, err := fnapi.Version(ctx)
-	if err != nil {
-		return nil, nil, fmt.Errorf("checking full node service API version: %w", err)
-	}
+	if doVersionCheck {
+		v, err := fnapi.Version(ctx)
+		if err != nil {
+			return nil, nil, fmt.Errorf("checking full node service API version: %w", err)
+		}
 
-	if !v.APIVersion.EqMajorMinor(lapi.FullAPIVersion1) {
-		return nil, nil, fmt.Errorf("full node service API version didn't match (expected %s, remote %s)", api.FullAPIVersion1, v.APIVersion)
+		if !v.APIVersion.EqMajorMinor(lapi.FullAPIVersion1) {
+			return nil, nil, fmt.Errorf("full node service API version didn't match (expected %s, remote %s)", api.FullAPIVersion1, v.APIVersion)
+		}
```

Comment:
Nit: this should probably be a different function altogether from `GetFullNodeApi`. `doVersionCheck` bool as a parameter is a bit weird.